PhysicsTools/NanoAOD/plugins/BaseMVAValueMapProducer.h

```diff
@@ -0,0 +1,159 @@
+#ifndef PhysicsTools_NanoAOD_BaseMVAValueMapProducer
+#define PhysicsTools_NanoAOD_BaseMVAValueMapProducer
+
+// -*- C++ -*-
+//
+// Package:    PhysicsTools/NanoAOD
+// Class:      BaseMVAValueMapProducer
+// 
+/**\class BaseMVAValueMapProducer BaseMVAValueMapProducer.cc PhysicsTools/NanoAOD/plugins/BaseMVAValueMapProducer.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Andre Rizzi
+//         Created:  Mon, 07 Sep 2017 09:18:03 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+
+#include "TMVA/Factory.h"
+#include "TMVA/Reader.h"
+
+#include "CommonTools/Utils/interface/StringObjectFunction.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "CommonTools/Utils/interface/TMVAZipReader.h"
+
+#include <string>
+//
+// class declaration
+//
+
+template <typename T>
+class BaseMVAValueMapProducer : public edm::stream::EDProducer<> {
```

Comment:
It would be useful to use a GlobalCache here, no? (to reduce total number of weight-file reads in multithreaded mode)
Similar to: https://github.com/cms-sw/cmssw/blob/master/RecoEgamma/EgammaTools/interface/MVAValueMapProducer.h